pkg/generate/app/cmd/newapp_test.go

```diff
@@ -415,6 +415,7 @@ func TestDetectSource(t *testing.T) {
 	dockerResolver := app.DockerRegistryResolver{
 		Client: dockerregistry.NewClient(),
 	}
+	mocks := app.MockSourceRepositories()
```

Comment:
this is just changing to use the same pointer for both expected and repository... in general, that's bad, because any mutation (even an invalid one) will still result in a match.